synapse/common.py

```diff
@@ -176,3 +184,18 @@ def worker(meth, *args, **kwargs):
     thr.setDaemon(True)
     thr.start()
     return thr
+
+@contextlib.contextmanager
+def tempdir():
```

Comment:
haha, almost this exact function is in synapse.tests.common.  would that one suffice?